frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -335,8 +337,9 @@ class Chart extends Component<Props, State> {
       this.chart.applyOptions({ timeScale: { timeVisible: false } });
     }
     this.setState(
-      { display: 'month', source: 'daily' },
+      { source: 'daily' },
       () => {
+        this.context.setChartDisplay('month');
         if (!this.chart) {
```

Comment:
Before this setstate callback ensured that both states were updated.

Afaik context function `setChartDisplay` is not synchronously. So when this callback runs the context not updated.

Now it seems that the intention is to fist update the state, and wait for context update.

Would it work if you set context before state?

That seems closer to how it was before, and React may have the chance to schedule the context and state update together?

Also it doesn't look like it should wait until state is updated before changing context. wdyt?

Reply:
Yeah the intention here is to actually only set the `setChartDisplay` after the `source` state has been changed.

I think setting context before state would also work, as you said it's closer to the previous behaviour (async, these two states not dependant on each other).

Reply:
Thinking about that we will at some point chagne this from class->functional I prefer if the context was set before the state.
Else it will look like it is intentional to wait for state change and then update context.

please change to set context first 🙏

Reply:
> please change to set context first

Yes, I'm changing it to setting the context state first. :)